samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -56,29 +56,42 @@ def prepare(params: dict) -> dict:
 
     try:
         # initialize terraform application
+        LOG.info("Initializing Terraform application")
         run(["terraform", "init"], check=True, capture_output=True)
 
         # get json output of terraform plan
+        LOG.info("Creating terraform plan and getting JSON output")
         with NamedTemporaryFile() as temp_file:
             run(["terraform", "plan", "-out", temp_file.name], check=True, capture_output=True)
             result = run(["terraform", "show", "-json", temp_file.name], check=True, capture_output=True)
         tf_json = json.loads(result.stdout)
 
         # convert terraform to cloudformation
+        LOG.info("Converting Terraform to CloudFormation")
         cfn_dict = _translate_to_cfn(tf_json)
+        LOG.info("Finished converting Terraform to CloudFormation")
 
         # store in supplied output dir
         if not os.path.exists(output_dir_path):
             os.mkdir(output_dir_path)
         metadataFilePath = os.path.join(output_dir_path, "template.json")
+        LOG.info("Storing CloudFormation in %s", metadataFilePath)
         with open(metadataFilePath, "w+") as metadata_file:
             json.dump(cfn_dict, metadata_file)
 
         return {"iac_applications": {"MainApplication": {"metadata_file": metadataFilePath}}}
 
-    except (CalledProcessError, OSError) as e:
+    except CalledProcessError as e:
         # one of the subprocess.run calls resulted in non-zero exit code or some OS error
+        LOG.debug(
+            "Error running terraform command: \n" "cmd: %s \n" "stdout: %s \n" "stderr: %s \n",
+            e.cmd,
+            e.stdout,
+            e.stderr,
+        )
         raise PrepareHookException("There was an error while preparing the Terraform application.") from e
```

Comment:
can we add unit test cases to cover the TF commands failures?

Reply:
there's `test_prepare_with_called_process_error`, or where you thinking of some other tests?

Reply:
sorry it is my bad, I missed it .. thanks Rup